hrqb/tasks/libhr_employee_appointments.py

```diff
@@ -0,0 +1,96 @@
+"""hrqb.tasks.libhr_employee_appointments"""
+
+import luigi  # type: ignore[import-untyped]
+import pandas as pd
+
+from hrqb.base.task import (
+    PandasPickleTask,
+    QuickbaseUpsertTask,
+)
+from hrqb.utils.quickbase import QBClient
+
+
+class ExtractLibHREmployeeAppointments(PandasPickleTask):
+    """Extract data from Library HR provided employee appointment data.
+
+    This task is expecting the CSV to be a local filepath.  Unlike other pipelines in this
+    client, this pipeline is rarely run, and is suitable for local, developer runs to load
+    data.
+    """
+
+    pipeline = luigi.Parameter()
+    stage = luigi.Parameter("Extract")
+    csv_filepath = luigi.Parameter()
+
+    def get_dataframe(self) -> pd.DataFrame:
+        return pd.read_csv(self.csv_filepath)
+
+
+class ExtractQBDepartments(PandasPickleTask):
+    """Get Departments data from Quickbase."""
+
+    pipeline = luigi.Parameter()
+    stage = luigi.Parameter("Extract")
+
+    def get_dataframe(self) -> pd.DataFrame:
+        qbclient = QBClient()
+        return qbclient.get_table_as_df(qbclient.get_table_id("Departments"))
+
+
+class TransformLibHREmployeeAppointments(PandasPickleTask):
+    """Enrich CSV data with data from other QB tables."""
+
+    stage = luigi.Parameter("Transform")
+    csv_filepath = luigi.Parameter()
+
+    def requires(self) -> list[luigi.Task]:  # pragma: nocover
+        return [
+            ExtractLibHREmployeeAppointments(
+                pipeline=self.pipeline, csv_filepath=self.csv_filepath
+            ),
+            ExtractQBDepartments(pipeline=self.pipeline),
+        ]
+
+    def get_dataframe(self) -> pd.DataFrame:
+        libhr_df = self.named_inputs["ExtractLibHREmployeeAppointments"].read()
+        departments_df = self.named_inputs["ExtractQBDepartments"].read()
```

Comment:
This is first time this has been used.  This task requires **two** parent tasks, therefore we cannot use `self.single_input_dataframe`.  This is utilizing the `self.named_inputs` to get the target data from the parent tasks by those task names.